scripts/fbi/hate_crime/preprocess.py

```diff
@@ -0,0 +1,357 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+import pandas as pd
+import json
+import numpy as np
+from utils import flatten_by_column, make_time_place_aggregation
+
+# Allows the following module imports to work when running as a script
+_SCRIPT_PATH = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(os.path.join(_SCRIPT_PATH,
+                             '../../../util/'))  # for statvar_dcid_generator
+
+from statvar_dcid_generator import get_statvar_dcid
+
+# Columns to input from source data
+INPUT_COLUMNS = [
+    'INCIDENT_ID', 'DATA_YEAR', 'OFFENDER_RACE', 'OFFENDER_ETHNICITY',
+    'STATE_ABBR', 'OFFENSE_NAME', 'BIAS_DESC', 'AGENCY_TYPE_NAME',
+    'MULTIPLE_OFFENSE', 'MULTIPLE_BIAS', 'PUB_AGENCY_NAME'
+]
+
+# Columns which do not contribute to a constraint property value in stat var
+NONPV_COLUMNS = [
+    'MULTIPLE_OFFENSE', 'MULTIPLE_BIAS', 'INCIDENT_ID', 'DATA_YEAR',
+    'STATE_ABBR', 'AGENCY_TYPE_NAME', 'PUB_AGENCY_NAME', 'LOCATION_NAME',
+    'BIAS_AGAINST'
+]
+
+# A dict to map bias descriptions to their category
+map_dict = {
```

Comment:
Is this still needed after adding biasMotivation in the config?

Reply:
Yes, this is needed to generate the aggregates for different Bias Categories (race, religion, gender)
Eg: biasMotivation: gender (Sum of male and female biasMotivations).